spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1812,8 +1811,8 @@ def _do_run(self, n_machine_time_steps, loading_done, run_until_complete):
         if self._use_virtual_board:
             logger.warn(
                 "Application will not actually be run as on a virtual board")
-        elif self._executable_start_type == \
-                ExecutableStartType.NO_APPLICATION:
+        elif (ExecutableType.NO_APPLICATION in
+                self._executable_types):
```

Comment:
Presumably I can't have more than one executable type if there is NO_APPLICATION.  Perhaps it should be an error if NO_APPLICATION is there with other types...

Reply:
probabily true. will recheck

Reply:
in this case, its output matching. the output is a dictonary. so it needs to chekc if its in the keys. it could also verify that length of keys is zero if thats what your going for. but i dont wnat to have multiple output tpyes, as itll complex the code even further